ted_sws/notice_validator/services/sparql_test_suite_runner.py

```diff
@@ -0,0 +1,128 @@
+
+TEMPLATES = Environment(loader=PackageLoader("ted_sws.notice_validator.resources", "templates"))
+SPARQL_TEST_SUITE_EXECUTION_HTML_REPORT_TEMPLATE = "sparql_query_results_report.jinja2"
+
+
+class SPARQLTestSuiteRunner:
+    """
+
+        One of the assumptions is that all the SPARQL queries are of type ASK.
+    """
+
+    def __init__(self, rdf_manifestation: RDFManifestation, sparql_test_suite: SPARQLTestSuite,
+                 mapping_suite: MappingSuite):
+        self.rdf_manifestation = rdf_manifestation
+        self.sparql_test_suite = sparql_test_suite
+        self.mapping_suite = mapping_suite
+
+    @classmethod
+    def _sparql_query_from_file_resource(cls, file_resource: FileResource) -> SPARQLQuery:
+        """
+        Gets file content and converts to a SPARQLQuery
+        :param file_resource:
+        :return:
+        """
+        file_content = StringIO(file_resource.file_content)
+        title = file_content.readline().split(":")[1].replace('\n', '').lstrip()
+        description = file_content.readline().split(":")[1].replace('\n', '').lstrip()
+        query = file_content.read()
+        return SPARQLQuery(title=title, description=description, query=query)
+
+    def execute_test_suite(self) -> SPARQLTestSuiteExecution:
+        """
+            Executing SPARQL queries from a SPARQL test suite and return execution details
+        :return:
+        """
+        sparql_runner = SPARQLRunner(self.rdf_manifestation.object_data)
+        test_suite_executions = SPARQLTestSuiteExecution(mapping_suite_identifier=self.mapping_suite.identifier,
+                                                         sparql_test_suite_identifier=self.sparql_test_suite.identifier,
+                                                         execution_results=[],
+                                                         object_data="SPARQLTestSuiteExecution")
+        for query_file_resource in self.sparql_test_suite.sparql_tests:
+            sparql_query = self._sparql_query_from_file_resource(file_resource=query_file_resource)
+            sparql_query_result = SPARQLQueryResult(query=sparql_query)
+            try:
+                query_result = sparql_runner.query(sparql_query.query)
+                sparql_query_result.result = str(
+                    query_result.askAnswer) if query_result.type == "ASK" else query_result.serialize(
+                    format="json")
+            except Exception as e:
+                sparql_query_result.error = str(e)
+            test_suite_executions.execution_results.append(sparql_query_result)
+        return test_suite_executions
+
+
+class SPARQLReportBuilder:
+    """
+        Given a SPARQLQueryResult, generates JSON and HTML reports.
+    """
+
+    def __init__(self, sparql_test_suite_execution: SPARQLTestSuiteExecution):
+        self.sparql_test_suite_execution = sparql_test_suite_execution
+
+    def generate_json(self) -> RDFValidationManifestation:
```

Comment:
this is misleading. We shall call a "report" only the HTML , whereas the internally it is the Domain objects that stand for JSON report.